aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,288 @@
+
+This route template would be handled by the `MvcRouteHandler <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Infrastructure/MvcRouteHandler/index.html>`_ ``IRouter`` instance. Tokens within curly braces (``{ }``) define `route value` parameters which will be bound if the route is matched. You can define more than one route value parameter in a route segment, but they must be separated by a literal value. For example ``{controller=Home}{action=Index}`` would not be a valid route, since there is no literal value between ``{controller}`` and ``{action}``. These route value parameters must have a name, and may have additional attributes specified.
+
+Route value parameters may have *default values*, designated by specifying the default after the parameter name, separated by an ``=``. For example, ``controller=Home`` would define ``Home`` as the default value for ``controller``. The default value is used if no value is present in the URL for the parameter. In addition to default values, route parameters may be optional (specified by appending a ``?`` to the end of the parameter name, as in ``id?``). The difference between optional and "has default" is that a route parameter with a default value always produces a value; an optional parameter may not. Route parameters may also have constraints, which further restrict which routes the template will match.
+
+The following table demonstrates some route template and their expected behavior.
+
+.. list-table:: Route Template Values
+	:header-rows: 1
+
+	* - Route Template
+	  - Example Matching URL
+	  - Notes
+	* - hello
+	  - /hello
+	  - Will only match the single path '/hello'
+	* - {Page=Home}
+	  - /
+	  - Will match and set ``Page`` to ``Home``.
+	* - {Page=Home}
+	  - /Contact
+	  - Will match and set ``Page`` to ``Contact``
+	* - {controller}/{action}/{id?}
+	  - /Products/List
+	  - Will map to ``Products`` controller and ``List`` method; Since ``id`` was not supplied in the URL, it's ignored.
+	* - {controller}/{action}/{id?}
+	  - /Products/Details/123
+	  - Will map to ``Products`` controller and ``List`` method, with ``id`` set to ``123``.
+	* - {controller=Home}/{action=Index}/{id?}
+	  - /
+	  - Will map to ``Home`` controller and ``Index`` method; ``id`` is ignored.
+
+.. _route-constraints:
+
+Route Constraints
+^^^^^^^^^^^^^^^^^
+	  
+Adding a colon ``:`` after the name allows additional inline constraints to be set on a route value parameter. Constraints with types always use the invariant culture - they assume the URL is non-localizable. Route constraints limit which URLs will match a route - URLs that do not match the constraint are ignored by the route. You can also use the ``*`` character as a prefix to a route value name to bind to the rest of the URI. For example, ``blog/{*slug}`` would match any URI that started with ``/blog/`` and had any value following it (which would be assigned to the ``slug`` route value.
```

Comment:
> You can also use the ``*`` character as a prefix . . .

This text about wildcard route parameters should be covered up above where you introduce template routes.